### PR TITLE
KOGITO-3135: adds new method for creating a release

### PR DESCRIPTION
**Pull request by mbiarnes**

### CODE REVIEW

File: vars/githubscm.groovy [hunk 1]
```diff
@@ -262,6 +262,27 @@ def removeRemoteTag(String remote, String tagName, String credentialsId = 'kie-c
...
+
+void createRelease(String tagName, String buildBranch, String credentialsId = 'kie-ci') {
+    def String description = "release ${tagName}"
+    withCredentials([usernamePassword(credentialsId: "${credentialsId}", usernameVariable: 'GH_USER', passwordVariable: 'GH_TOKEN')]) {
+        //check if release already exist
+        if(releaseExists(tagName)){
+            echo "release ${tagName} already exist"
+            echo "removing release ${tagName}"
+            sh "gh release delete ${tagName} -y"
+        }
```
Comment (radtriste) [inline vars/githubscm.groovy:281]: please create a `deleteRelease` method and let the "user" decide

Comment (radtriste) [reply vars/githubscm.groovy:281]: *decide whether it wants to delete if already existing

File: vars/githubscm.groovy [hunk 2]
```diff
@@ -262,6 +262,30 @@ def removeRemoteTag(String remote, String tagName, String credentialsId = 'kie-c
     println "[INFO] Deleted remote tag ${tagName}."
 }
 
+/*
+* Checks if a release already exists, removes it and creates a new
```
Comment (radtriste) [inline vars/githubscm.groovy:266]: please update comment with new meaning

File: vars/githubscm.groovy [hunk 3]
```diff
@@ -262,6 +262,30 @@ def removeRemoteTag(String remote, String tagName, String credentialsId = 'kie-c
     println "[INFO] Deleted remote tag ${tagName}."
 }
 
+/*
+* Checks if a release already exists, removes it and creates a new 
+*/
+
+void createRelease(String tagName, String buildBranch, String credentialsId = 'kie-ci') {
+    def String description = "release ${tagName}"
```
Comment (radtriste) [inline vars/githubscm.groovy:270]:
```suggestion
    String description = "release ${tagName}"
```

Comment (radtriste) [reply vars/githubscm.groovy:270]: Also can we make the description a parameter of the method, with default to `Release ${tagName}`

File: vars/githubscm.groovy [hunk 4]
```diff
@@ -262,6 +262,30 @@ def removeRemoteTag(String remote, String tagName, String credentialsId = 'kie-c
...
+    withCredentials([usernamePassword(credentialsId: "${credentialsId}", usernameVariable: 'GH_USER', passwordVariable: 'GH_TOKEN')]) {
+        sh "gh release create ${tagName} --target ${buildBranch} --title ${tagName} --notes \"${description}\""
+    }
+}
+
+void deleteRelease(String tagName, String credentialsId = 'kie-ci') {
+    withCredentials([usernamePassword(credentialsId: "${credentialsId}", usernameVariable: 'GH_USER', passwordVariable: 'GH_TOKEN')]) {        
+        echo "removing release ${tagName}"
```
Comment (radtriste) [inline vars/githubscm.groovy:278]: Useless I would say
```suggestion
```

File: vars/githubscm.groovy [hunk 5]
```diff
@@ -262,6 +262,30 @@ def removeRemoteTag(String remote, String tagName, String credentialsId = 'kie-c
...
+        sh "gh release delete ${tagName} -y"
+    }
+}
+
+boolean isReleaseExist(String tagName) {
+    // checks if the release is already existing
+    status = sh(script: "gh release view ${tagName}", returnStatus: true)
+    return status == 0
```
Comment (radtriste) [inline vars/githubscm.groovy:286]:
```suggestion
    return sh(script: "gh release view ${tagName}", returnStatus: true) == 0
```

**Review by Ginxo — CHANGES_REQUESTED** [review]
please provide tests for isReleaseExist

**Comment by mbiarnes** [comment]
@Ginxo 
Done. Tests provided.

**Review by Ginxo — APPROVED** [review]
LGTM

File: vars/githubscm.groovy [hunk 6]
```diff
@@ -262,6 +262,34 @@ def removeRemoteTag(String remote, String tagName, String credentialsId = 'kie-c
     println "[INFO] Deleted remote tag ${tagName}."
 }
 
+/*
+* Creates a new release on GitHub 
+*/
+
```
Comment (radtriste) [inline vars/githubscm.groovy:268]:
```suggestion
```

File: vars/githubscm.groovy [hunk 7]
```diff
@@ -262,6 +262,34 @@ def removeRemoteTag(String remote, String tagName, String credentialsId = 'kie-c
...
+void createRelease(String tagName, String buildBranch, String description = "Release ${tagName}", String credentialsId = 'kie-ci') {
+    withCredentials([usernamePassword(credentialsId: "${credentialsId}", usernameVariable: 'GH_USER', passwordVariable: 'GH_TOKEN')]) {
+        sh "gh release create ${tagName} --target ${buildBranch} --title ${tagName} --notes \"${description}\""
+    }
+}
+
+/*
+* Removes a release on GitHub
+*/
+
```
Comment (radtriste) [inline vars/githubscm.groovy:278]:
```suggestion
```

File: vars/githubscm.groovy [hunk 8]
```diff
@@ -262,6 +262,34 @@ def removeRemoteTag(String remote, String tagName, String credentialsId = 'kie-c
...
+void deleteRelease(String tagName, String credentialsId = 'kie-ci') {
+    withCredentials([usernamePassword(credentialsId: "${credentialsId}", usernameVariable: 'GH_USER', passwordVariable: 'GH_TOKEN')]) {
+        sh "gh release delete ${tagName} -y"
+    }
+}
+
```
Comment (radtriste) [inline vars/githubscm.groovy:284]:
```suggestion
/*
* Checks whether a release exists on GitHub
*/
```

File: vars/githubscm.groovy [hunk 9]
```diff
@@ -262,6 +262,34 @@ def removeRemoteTag(String remote, String tagName, String credentialsId = 'kie-c
...
+boolean isReleaseExist(String tagName, String credentialsId = 'kie-ci') {
+    withCredentials([usernamePassword(credentialsId: "${credentialsId}", usernameVariable: 'GH_USER', passwordVariable: 'GH_TOKEN')]) {
+        // checks if the release is already existing
+        exist =  sh(script: "gh release view ${tagName}", returnStatus: true) == 0
```
Comment (radtriste) [inline vars/githubscm.groovy:288]:
```suggestion
        exist = sh(script: "gh release view ${tagName}", returnStatus: true) == 0
```

**Review by radtriste — APPROVED** [review]

**Pull request merged**